server/utils/utils.test.ts

```diff
@@ -317,3 +318,14 @@ describe('zip', () => {
     ])
   })
 })
+
+describe('isValidDate', () => {
+  it('has invalid month', () => {
+    const result = isValidDate('01', '42', '2000')
+    expect(result).toEqual(false)
+  })
+  it('has valid date', () => {
+    const result = isValidDate('01', '05', '2000')
+    expect(result).toEqual(true)
+  })
```

Comment:
be good to have cases for both when no date fields are present and some date fields are present

Reply:
added